docs/proposals/central-segmented-registry.md

```diff
@@ -0,0 +1,144 @@
+
+Today, OpenShift deployments support a single integrated registry, referenced as service **docker-registry** in the **default** namespace. All other registries are "external". This proposal suggests a way to achieve more flexibility in referencing registries.
+
+## Goals
+
+* **Consistent Image Reference** We want users to consistently reference the same image name across all of their clusters as part of a promotion workflow. We want to do this through controlling environment assignment at the cluster level. In other words, the production cluster should only have access to images in the production registry.
+* **External Registry Abstraction** We also want users to be able to reference "external" registries as an integrated service abstraction, regardless of the actual cluster the registry is hosted.
+* **Optional with Backward Compatibility** This proposes significant complexity. Using it should be optional. Administrators and users who do not require this feature should not have to do anything differently.
+* **On-Premise** This proposal is focused on solving issues unique to private, on-premise users of OpenShift. Untrusted, multi-tenant use is not in scope.
+
+## Central Registry Support
+
+For the purposes of discussion, a "central registry" is a simply a dedicated OpenShift cluster for serving images. As described below, this "registry cluster" hosts one or more registry deployments to support environment segmentation. The registry cluster may serve one or many OpenShift clusters.
+
+## Segmenting Registries
+
+The OpenShift registry is reference by `<project>/<registry-service>`. The default registry deployment is `default/docker-registry`. Multiple registries may be deployed on the same cluster using unique project namespaces. Each registry is addressed using a unique route.
+
+### Managing Images
+
+ImageStreamTags are annotated (annotations or labels?) with a list of approved registry environments. Users may promote (or demote) images by managing the environment whitelist. For backward compatibility, the default environment is "default". For example:
+
+        kind: ImageStreamTag
+          annotations:
+            environment:
+              - test
+              - uat
+
+To promote to the production environment, a user or process adds an annotation to the ImageStreamTag object. This action would trigger a job to manage ImageStream metadata, registry and storage so the image could be served from the desired registry environment.
+
+        kind: ImageStreamTag
+          annotations:
+            environment:
+              - test
+              - uat
+              - prod
+
+### Managing the Registries
+
+The central registry cluster may have one or many registries deployed. Each registry is deployed into a project matching the environment name (e.g. test, uat, prod). For example:
```

Comment:
Why does the project have to match the environment name?  Just a convention?  Seems arbitrary.